### PR TITLE
#LANG-1114 TypeUtils.equals is broken for WildcardType

### PR DESCRIPTION
**Pull request by datalorax**
TypeUtils.equals is broken for WildcardType, when compared with non-wildcard type.

It was broken because it was returning true! Now returns false. Also added test to cover the issue.

### CODE REVIEW

File: src/changes/changes.xml [hunk 1]
```diff
@@ -22,6 +22,7 @@
   <body>
 
   <release version="3.5" date="tba" description="tba">
+    <action issue="LANG-1114" type="fix" dev="acoates">Fix bug in TypeUtils.equals where comparison of wildcard and nom-wildcard type was incorrectly returning true</action>
```
Comment (britter) [inline src/changes/changes.xml]: Usually we use the dev attribute for the committer who reviews the patch. Contributors are attributed in the due-to attribute. I hope it is okay for you if I modify this when I apply this fix.

Comment (datalorax) [inline src/changes/changes.xml]: OK course. np.

**Comment by britter** [comment]
@datalorax Commons Lang has been migrated from svn to git. Because of this the main branch has been renamed to 'master'. Can you please rebase your work against the master branch? Thank you!

**Comment by britter** [comment]
Hello @datalorax it looks like rebasing did not work the way we wanted. The diff shows more then 500 touched filed. Maybe it's easier to close this PR and recreate it against the master branch. Sorry for the inconvenience!

**Comment by datalorax** [comment]
Sorry (this is a bit embarrassing!), I was just playing around with the rebase and didn't mean to raise a PR against this.

**Pull request closed without merging**